cads_adaptors/tools/download_tools.py

```diff
@@ -4,22 +4,26 @@
 from cads_adaptors.tools.general import ensure_list
 
 
-# TODO use targzstream
+# TODO zipstream for archive creation
 def zip_paths(paths: List[str], base_target: str = "output-data", **kwargs) -> BinaryIO:
     import zipfile
 
     target = f"{base_target}.zip"
     with zipfile.ZipFile(target, mode="w") as archive:
         for p in paths:
-            archive.write(p)
+            if kwargs.get("preserve_dir", False):
+                arcname = p
+            else:
+                arcname = os.path.basename(p)
+            archive.write(p, arcname)
```

Comment:
It took me a moment to work out what the variables were here because of the abbreviated names - could we change them to something like:
`p -> path`
`arcname -> archive_name`